modules/database/export.bal

```diff
@@ -272,40 +279,89 @@ public isolated transactional function getTimelineStepLimit(int experimentId) re
+        stepList = config.stepList.sort();
+    }
+    int counter = 0;
+    boolean add;
     foreach TimelineStepFull timelineStepDb in timelineStepListDb {
-        TimelineStepExport timelineStepExport = check castToTimelineStepExport(timelineStepDb);
-        // retrieve associated step data
-        timelineStepExport.stepDataList = check getTimelineStepDataList(timelineStepDb.stepId);
-        foreach var stepData in timelineStepExport.stepDataList {
-            dataIdList.push(stepData.dataId);
+        add = true;
+        if config.restriction == "STEPS" && timelineStepDb.sequence == stepList[counter] {
+            // filter steps with stepList 
+            if timelineStepDb.sequence == stepList[counter] {
+                counter += 1;
+            } else {
+                add = false;
+            }
         }
-        // retrieve associated substeps with their substep data
-        timelineStepExport.timelineSubsteps = check getTimelineSubstepsExport(timelineStepDb.stepId);
-        timelineSteps.push(timelineStepExport);
-        foreach var substep in timelineStepExport.timelineSubsteps {
-            foreach var substepData in substep.substepDataList {
-                dataIdList.push(substepData.dataId);
+        if add {
+            TimelineStepExport timelineStepExport = check castToTimelineStepExport(timelineStepDb);
+            if config.restriction != "LOGS" {
+                // retrieve associated step data
+                timelineStepExport.stepDataList = check getTimelineStepDataList(timelineStepDb.stepId);
+                foreach var stepData in timelineStepExport.stepDataList {
+                    dataIdList.push(stepData.dataId);
+                }
+            } else {
+                // don't need data for restriction "LOGS"
+                timelineStepExport.stepDataList = [];
             }
+            // retrieve associated substeps with their substep data
+            timelineStepExport.timelineSubsteps = check getTimelineSubstepsExport(timelineStepDb.stepId, config);
+            timelineSteps.push(timelineStepExport);
+            if config.restriction != "LOGS" {
+                // retrieve associated substep data
+                foreach var substep in timelineStepExport.timelineSubsteps {
+                    foreach var substepData in substep.substepDataList {
+                        dataIdList.push(substepData.dataId);
+                    }
+                }
+            } // else don't need data
         }
     }
 
-    // ignore duplicates in dataIdList and create experiment data list
-    int[] sortedDataIdList = dataIdList.sort();
-    int tmp = -1;
-    foreach int dataId in sortedDataIdList {
-        if dataId != tmp {
-            tmp = dataId;
-            ExperimentDataExport experimentData = check getExperimentDataExport(experimentId, dataId);
-            experimentDataList.push(experimentData);
+    if config.restriction != "LOGS" {
```

Comment:
Make this check test if there are any data ids to be exported (or just always do it since there is no harm doing it for an empty list). Maybe refactor this out into its own method.